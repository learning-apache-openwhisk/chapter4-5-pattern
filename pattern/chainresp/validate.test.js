const ow = require("openwhisk")
//const ow = openwhisk()

function inv(par) {
  return {
    name: "pattern/chainresp-validate",
    params: par
  };
}

let p0 = {};
let p1 = { name: "Michele", ...p0 };
let p2 = { email: "michele@sciabarra.com", ...p1 };
let p3 = { phone: "1234567890", ...p2 };
let call = "pattern/chainresp-validate";


test("chainresp-validate0", () =>
  ow()
    .actions.invoke(inv(p0))
    .then(res => expect(res).toMatchSnapshot()));

test("chainresp-validate1", () =>
  ow()
    .actions.invoke(inv(p1))
    .then(res => expect(res).toMatchSnapshot()));

test("chainresp-validate2", () =>
  ow()
    .actions.invoke(inv(p2))
    .then(res => expect(res).toMatchSnapshot()));

    test("chainresp-validate3", () =>
  ow()
    .actions.invoke(inv(p3))
    .then(res => expect(res).toMatchSnapshot()));

