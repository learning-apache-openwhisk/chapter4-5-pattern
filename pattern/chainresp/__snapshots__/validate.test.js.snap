// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chainresp-validate0 1`] = `
Object {
  "errors": Array [
    "missing name",
    "missing email",
    "missing phone",
  ],
  "message": Array [],
}
`;

exports[`chainresp-validate1 1`] = `
Object {
  "errors": Array [
    "missing email",
    "missing phone",
  ],
  "message": Array [
    "name: Michele",
  ],
  "name": "Michele",
}
`;

exports[`chainresp-validate2 1`] = `
Object {
  "email": "michele@sciabarra.com",
  "errors": Array [
    "missing phone",
  ],
  "message": Array [
    "name: Michele",
    "email: michele@sciabarra.com",
  ],
  "name": "Michele",
}
`;

exports[`chainresp-validate3 1`] = `
Object {
  "email": "michele@sciabarra.com",
  "errors": Array [],
  "message": Array [
    "name: Michele",
    "email: michele@sciabarra.com",
    "phone: 1234567890",
  ],
  "name": "Michele",
  "phone": "1234567890",
}
`;
